spec/update-process-env-spec.js

```diff
@@ -124,30 +182,42 @@ describe('updateProcessEnv(launchEnv)', function () {
     })
 
     describe('shouldGetEnvFromShell()', function () {
-      it('returns the shell when the shell should be patched', function () {
+      it('indicates when the environment should be fetched from the shell', function () {
         process.platform = 'darwin'
-        expect(shouldGetEnvFromShell('/bin/sh')).toBe(true)
-        expect(shouldGetEnvFromShell('/usr/local/bin/sh')).toBe(true)
-        expect(shouldGetEnvFromShell('/bin/bash')).toBe(true)
-        expect(shouldGetEnvFromShell('/usr/local/bin/bash')).toBe(true)
-        expect(shouldGetEnvFromShell('/bin/zsh')).toBe(true)
-        expect(shouldGetEnvFromShell('/usr/local/bin/zsh')).toBe(true)
-        expect(shouldGetEnvFromShell('/bin/fish')).toBe(true)
-        expect(shouldGetEnvFromShell('/usr/local/bin/fish')).toBe(true)
+        expect(shouldGetEnvFromShell({SHELL: '/bin/sh'})).toBe(true)
+        expect(shouldGetEnvFromShell({SHELL: '/usr/local/bin/sh'})).toBe(true)
+        expect(shouldGetEnvFromShell({SHELL: '/bin/bash'})).toBe(true)
+        expect(shouldGetEnvFromShell({SHELL: '/usr/local/bin/bash'})).toBe(true)
+        expect(shouldGetEnvFromShell({SHELL: '/bin/zsh'})).toBe(true)
+        expect(shouldGetEnvFromShell({SHELL: '/usr/local/bin/zsh'})).toBe(true)
+        expect(shouldGetEnvFromShell({SHELL: '/bin/fish'})).toBe(true)
+        expect(shouldGetEnvFromShell({SHELL: '/usr/local/bin/fish'})).toBe(true)
+        process.platform = 'linux'
+        expect(shouldGetEnvFromShell({SHELL: '/bin/sh'})).toBe(true)
+        expect(shouldGetEnvFromShell({SHELL: '/usr/local/bin/sh'})).toBe(true)
+        expect(shouldGetEnvFromShell({SHELL: '/bin/bash'})).toBe(true)
+        expect(shouldGetEnvFromShell({SHELL: '/usr/local/bin/bash'})).toBe(true)
+        expect(shouldGetEnvFromShell({SHELL: '/bin/zsh'})).toBe(true)
+        expect(shouldGetEnvFromShell({SHELL: '/usr/local/bin/zsh'})).toBe(true)
+        expect(shouldGetEnvFromShell({SHELL: '/bin/fish'})).toBe(true)
+        expect(shouldGetEnvFromShell({SHELL: '/usr/local/bin/fish'})).toBe(true)
       })
 
       it('returns false when the shell should not be patched', function () {
```

Comment:
Maybe this should be phrased `returns false when the environment indicates that Atom was launched from a shell`